test/images/serve_hostname/Makefile

```diff
@@ -12,26 +12,90 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-all: serve_hostname
+# Cross-build the serve_hostname image
+#
+# Usage:
+#       [TAG=v1.5] [PREFIX=gcr.io/google_containers] [TEST_REGISTRY=b.gcr.io/k8s_authenticated_test] [ARCH=amd64] [BASEIMAGE=busybox] make all
+
+.PHONY: all push container clean
+
+TAG ?= v1.5
+
+REGISTRY ?= gcr.io/google_containers
+TEST_REGISTRY ?= b.gcr.io/k8s_authenticated_test
+
+# Architectures supported: amd64, arm, arm64 and ppc64le
+ARCH ?= amd64
+
+ALL_ARCH = amd64 arm arm64 ppc64le
+
+GOARM=6
+TEMP_DIR := $(shell mktemp -d)
+KUBE_CROSS_IMAGE = gcr.io/google_containers/kube-cross
+KUBE_CROSS_VERSION = $(shell cat ../../../build/build-image/cross/VERSION)
+
+BIN = serve_hostname
+SRCS = serve_hostname.go
+
+IMAGE = $(REGISTRY)/$(BIN)-$(ARCH)
+TEST_IMAGE = $(TEST_REGISTRY)/$(BIN)-$(ARCH)
 
-TAG = v1.4
-PREFIX = gcr.io/google_containers
-TEST_PREFIX = b.gcr.io/k8s_authenticated_test
+# Set default base image dynamically for each arch
+ifeq ($(ARCH),amd64)
+        BASEIMAGE?=busybox
+endif
+ifeq ($(ARCH),arm)
+        BASEIMAGE?=armel/busybox
+endif
+ifeq ($(ARCH),arm64)
+        BASEIMAGE?=aarch64/busybox
+endif
+ifeq ($(ARCH),ppc64le)
+        BASEIMAGE?=ppc64le/busybox
+endif
 
-serve_hostname: serve_hostname.go
-	CGO_ENABLED=0 go build -a -installsuffix cgo --ldflags '-w' ./serve_hostname.go
+# If you want to build AND push all containers, see the 'all-push' rule.
+all: all-container
 
-container: serve_hostname
-	docker build -t $(PREFIX)/serve_hostname:$(TAG) .
-	if [ -n "$(TEST_PREFIX)" ]; then \
-	    docker tag -f $(PREFIX)/serve_hostname:$(TAG) $(TEST_PREFIX)/serve_hostname:$(TAG); \
+sub-container-%:
+	$(MAKE) ARCH=$* container
+
+sub-push-%:
+	$(MAKE) ARCH=$* push
+
+all-container: $(addprefix sub-container-,$(ALL_ARCH))
+
+all-push: $(addprefix sub-push-,$(ALL_ARCH))
+
+build: bin/$(BIN)-$(ARCH)
+
+bin/$(BIN)-$(ARCH): $(SRCS)
+	# Copy the content in this dir to the temp dir
+	cp ./* $(TEMP_DIR)
+
+	docker run -it -v $(TEMP_DIR):/build \
+	$(KUBE_CROSS_IMAGE):$(KUBE_CROSS_VERSION) \
```

Comment:
golang:$(GOLANG_VERSION) is just enough here, you can set GOLANG_VERSION to 1.6.3